pkg/didcomm/transport/http/tls/certpool.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package tls
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"log"
+	"sync"
+	"sync/atomic"
+)
+
+// certPool is a thread safe wrapper around the x509 standard library
+// cert pool implementation.
+// It optionally allows loading the system trust store.
+type certPool struct {
+	certPool       *x509.CertPool
+	certs          []*x509.Certificate
+	certsByName    map[string][]int
+	lock           sync.RWMutex
+	dirty          int32
+	systemCertPool bool
+}
+
+// NewCertPool new CertPool implementation
+func NewCertPool(useSystemCertPool bool) (CertPool, error) {
```

Comment:
@Baha-sk why not return a pointer?

Reply:
@Baha-sk any reason why the user cannot simply provide their own certpool? Seems pretty easy to do, and I'd argue that if a user wants a custom cert pool then they must know what they're doing.

Having the user pass in a cert pool to the constructor function in the transport layer would void all of this code to avoid memory inconsistencies (swapping cert pools, mutexes, filtering duplicates, etc. etc)

Reply:
this can be done in a separate PR, we can add another options to the new outbound client that accepts a cert pool instance, path or no options if they wish to use the system cert pool.. 


this initial HTTP client Agent PR assumes the user of the library will pass in a path to the certpool, enhancements can be added on top of it.

Reply:
@Baha-sk we incur additional maintenance by adding all these unnecessary features. For now, let's just have the user provide their own cert pool and ff they don't, we can reasonably use the system cert pool by default. Right now it's not clear if we need dynamic cert pools.